add.go

```diff
@@ -161,6 +170,12 @@ func getURL(src string, chown *idtools.IDPair, mountpoint, renameTarget string,
 		return fmt.Errorf("writing content from %q to tar stream: %w", src, err)
 	}
 
+	if digester != nil {
+		if responseDigest := digester.Digest(); responseDigest != srcDigest {
+			return fmt.Errorf("unexpected response digest: %s, want %s", responseDigest, srcDigest)
```

Comment:
Please add the URL to this error message, as it could be one of multiple items referenced in an ADD instruction.